src/index.jsx

```diff
@@ -11,53 +11,73 @@ const documentScrollPositionKey = {
   x: 'scrollLeft'
 }
 
-const getScrollPosition = (axis) =>
+const elementScrollPositionKey = documentScrollPositionKey
+
+const getScrollPositionWindow = (axis) =>
   window[windowScrollPositionKey[axis]] ||
   document.documentElement[documentScrollPositionKey[axis]] ||
   document.body[documentScrollPositionKey[axis]] ||
   0
 
+const getScrollPositionElement = (ref, axis) =>
+  ref[elementScrollPositionKey[axis]]
+
+const getScrollPosition = (ref, axis) => {
+  if (ref === window) {
+    return getScrollPositionWindow(axis)
+  }
+  return getScrollPositionElement(ref, axis)
+}
+
 export const ReactWindowScroller = ({
   children,
   throttleTime = 10,
-  isGrid = false
+  isGrid = false,
+  scrollElementRef = window
 }) => {
   const ref = useRef()
   const outerRef = useRef()
 
   useEffect(() => {
     const handleWindowScroll = throttle(() => {
       const { offsetTop = 0, offsetLeft = 0 } = outerRef.current || {}
-      const scrollTop = getScrollPosition('y') - offsetTop
-      const scrollLeft = getScrollPosition('x') - offsetLeft
+
+      const scrollTop = getScrollPosition(scrollElementRef, 'y') - offsetTop
+      const scrollLeft = getScrollPosition(scrollElementRef, 'x') - offsetLeft
       if (isGrid) ref.current && ref.current.scrollTo({ scrollLeft, scrollTop })
       if (!isGrid) ref.current && ref.current.scrollTo(scrollTop)
     }, throttleTime)
 
-    window.addEventListener('scroll', handleWindowScroll)
+    scrollElementRef.addEventListener('scroll', handleWindowScroll)
     return () => {
       handleWindowScroll.cancel()
-      window.removeEventListener('scroll', handleWindowScroll)
+      scrollElementRef.removeEventListener('scroll', handleWindowScroll)
     }
-  }, [isGrid])
+  }, [isGrid, scrollElementRef, throttleTime])
 
   const onScroll = useCallback(
-    ({ scrollLeft, scrollTop, scrollOffset, scrollUpdateWasRequested }) => {
+    ({
+      scrollLeft = 0, // This is not provided by react-window
+      scrollTop = 0, // This is not provided by react-window
```

Comment:
`react-window` passes `scrollDirection: ScrollDirection` not `scrollLeft` and `scrollTop`. Without setting `0` as default, this function does not do anything ass the numbers end up as `NaN`. From what I can tell, defaulting to `0` resolves that issue and results in this executing properly. Not sure why they are part of the method signature, though.

https://github.com/bvaughn/react-window/blob/master/src/createListComponent.js#L393-L409